### ISSUE TITLE
Add ability to update offer

### ISSUE THREAD

**Issue by sbddesign** [ATLBitLab/twelvecash#7]
Ability to update the offer, which requires a way to authenticate the user. Probably rely on a signed message from the lightning node/wallet. A message is signed when username is created, then we expect a valid signature from the same pubkey in order to update the offer?

Another tricky thing is that if the lightning node/wallet changes, then you might want to change the offer still without having access to the original key. If I lose access to a lightning wallet (I hope not), I still want to own `stephen` and `sbddesign`.

**Comment by paulosacramento** [comment]
I have been thinking a lot about this. VPN services that do not require user credentials for registration may be the best example of how to deal with this problem. All the other alternatives (email, lightning wallet, nostr, etc) seem not be suitable.

Mullvad's example can be followed here: during the registration process on Twelve Cash, an _Secret Account Number_ could be generated. This _Secret Account Number_ would act both as an identifier in the context of the Twelve Cash system and as a password for logging in and making changes.

As we can see in the screenshot below, it would also be very important to clearly communicate that this information should be kept private.
![Screenshot 2024-08-31 at 16 54 31](https://github.com/user-attachments/assets/2f1bcb1a-3173-457f-8deb-05566e7a0259)

**Comment by sbddesign** [comment]
Yeah, a secret account number could be good. However, we have already implemented a Nostr login. So it seems to me like the lowest hanging fruit is just to require a login. @chdwlch and I have discussed email auth as well.

But I dunno, the secret number idea would be easy to implement. It's not a bad idea. Hmm.
